packages/expo-device/ios/UIDevice.swift

```diff
@@ -0,0 +1,199 @@
+            case "iPhone11,2":                                    return DeviceMapping(modelName: "iPhone XS", deviceYearClass: 2018)
+            case "iPhone11,4", "iPhone11,6":                      return DeviceMapping(modelName: "iPhone XS Max", deviceYearClass: 2018)
+            case "iPhone11,8":                                    return DeviceMapping(modelName: "iPhone XR", deviceYearClass: 2018)
+            case "iPhone12,1":                                    return DeviceMapping(modelName: "iPhone 11", deviceYearClass: 2019)
+            case "iPhone12,3":                                    return DeviceMapping(modelName: "iPhone 11 Pro", deviceYearClass: 2019)
+            case "iPhone12,5":                                    return DeviceMapping(modelName: "iPhone 11 Pro Max", deviceYearClass: 2019)
+            case "iPhone13,1":                                    return DeviceMapping(modelName: "iPhone 12 mini", deviceYearClass: 2020)
+            case "iPhone13,2":                                    return DeviceMapping(modelName: "iPhone 12", deviceYearClass: 2020)
+            case "iPhone13,3":                                    return DeviceMapping(modelName: "iPhone 12 Pro", deviceYearClass: 2020)
+            case "iPhone13,4":                                    return DeviceMapping(modelName: "iPhone 12 Pro Max", deviceYearClass: 2020)
+            case "iPhone14,4":                                    return DeviceMapping(modelName: "iPhone 13 mini", deviceYearClass: 2021)
+            case "iPhone14,5":                                    return DeviceMapping(modelName: "iPhone 13", deviceYearClass: 2021)
+            case "iPhone14,2":                                    return DeviceMapping(modelName: "iPhone 13 Pro", deviceYearClass: 2021)
+            case "iPhone14,3":                                    return DeviceMapping(modelName: "iPhone 13 Pro Max", deviceYearClass: 2021)
+            case "iPhone14,7":                                    return DeviceMapping(modelName: "iPhone 14", deviceYearClass: 2022)
+            case "iPhone14,8":                                    return DeviceMapping(modelName: "iPhone 14 Plus", deviceYearClass: 2022)
+            case "iPhone15,2":                                    return DeviceMapping(modelName: "iPhone 14 Pro", deviceYearClass: 2022)
+            case "iPhone15,3":                                    return DeviceMapping(modelName: "iPhone 14 Pro Max", deviceYearClass: 2022)
+            case "iPhone8,4":                                     return DeviceMapping(modelName: "iPhone SE", deviceYearClass: 2016)
+            case "iPhone12,8":                                    return DeviceMapping(modelName: "iPhone SE (2nd generation)", deviceYearClass: 2020)
+            case "iPhone14,6":                                    return DeviceMapping(modelName: "iPhone SE (3rd generation)", deviceYearClass: 2022)
```

Comment:
Don't you guys think it's very strange that the expo-device version of the iphone is different from the real name of the iPhone model?

Reply:
> Don't you guys think it's very strange that the expo-device version of the iphone is different from the real name of the iPhone model?

If you mean by strings like `iPhone14,7`, this is kind of the expected way to gather this type of data. Apple does not expose a one-step way to get the exact consumer-facing model name of the iPhone, so what is done instead is - you first need to get a model identifier and then map it to the corresponding model name. See more info here: https://stackoverflow.com/a/26962452